aiohttp_sentry/__init__.py

```diff
@@ -1,22 +1,49 @@
+import asyncio
 from functools import partial
+import sys
 
 import raven
 import raven_aiohttp
 
 
 class SentryMiddleware:
 
-    def __init__(self, sentry_kwargs=None):
+    def __init__(self, install_excepthook=True, sentry_kwargs=None):
         if sentry_kwargs is None:
             sentry_kwargs = {}
 
+        if install_excepthook:
+            # do not let raven.Client install its own excepthook
+            sentry_kwargs['install_sys_hook'] = False
+
         sentry_kwargs = {
             'transport': raven_aiohttp.AioHttpTransport,
             'enable_breadcrumbs': False,
             **sentry_kwargs,
         }
         self.client = raven.Client(**sentry_kwargs)
 
+        if install_excepthook:
+            self.update_excepthook()
+
+    def update_excepthook(self):
+        """Update sys.excepthook so that it closes the sentry transport."""
+
+        loop = asyncio.get_event_loop()
```

Comment:
I'm sorry, but I'm scared to merge it like this. I'm not familiar with `asyncio` to understand why it's done like this, but pretty much everywhere I see the `loop` being asked for as a parameter, and defaulting to `asyncio.get_event_loop()` only if it's not set.

So this makes me feel like if we don't allow the users to pass their own `loop`, we might do unexpected things for some users.

Could you either
- explain why this is fine here anyway,
- or allow users to override the loop that we use here
? Thanks!

Reply:
A good point. I am not too familiar with `asyncio` either, but it seems reasonable to allow using custom event loop even for closing the Sentry transport. I have changed the code, so that a custom event loop can be passed in as parameter.